File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        n=int(len(prices))
        mp=0
        bb=prices[0]
        i=1
        for i in range(n):
            if prices[i]>bb:
                mp=max(mp,prices[i]-bb)
            bb=min(bb,prices[i])
        return mp

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int mp=0,bb=prices[0],n=prices.length;
        for(int i=1;i<n;i++)
        {
            if(prices[i]>bb)
            {
                mp=Math.max(mp,prices[i]-bb);
            }
            bb=Math.min(bb,prices[i]);
        }
        return mp;
    }
}